adagrams/game.py

```diff
@@ -1,13 +1,168 @@
+        'P': 2, 'Q': 1, 'R': 6, 'S': 4, 'T': 6, 
+        'U': 4, 'V': 2, 'W': 2, 'X': 1, 'Y': 2, 'Z': 1
+    }
+    
+    # list of all letters
+    all_letters = []
+    for letter, count in letter_pool.items():
+        all_letters.extend([letter] * count)
+    
+    # Draw 10 random letters from list
+    hand = []
+    remaining_letters = all_letters.copy() 
+    
+    for i in range(10):
+        if remaining_letters:
+            draw_position = randint(0, len(remaining_letters) - 1)
+            selected_letter = remaining_letters.pop(draw_position)
+            hand.append(selected_letter)
+    
+    return hand
+
 
 def uses_available_letters(word, letter_bank):
-    pass
+    # Convert word to uppercase
+    word = word.upper()
+    
+    # Increcrement count of letters or start count
+    available_letters = {}
+    for letter in letter_bank:
+        if letter in available_letters:
+            available_letters[letter] += 1
+        else:
+            available_letters[letter] = 1
+    
+    # Check if we have this letter or have used them all
+    for letter in word:
+        if letter not in available_letters or available_letters[letter] == 0:
+            return False
+        
+        # Reduce avalibility count 
+        available_letters[letter] -= 1
+    
+    # Get through whole word without running out of letters for word to be vaild
+    return True
 
 def score_word(word):
-    pass
+
+    # Convert word to uppercase
+    word = word.upper()
+
+    # Handle empty word case
+    if not word:
+        return 0
+    
+    # Mapp letters to point value
+    point_values = {
+        1: "A, E, I, O, U, L, N, R, S, T",
+        2: "D, G",
+        3: "B, C, M, P",
+        4: "F, H, V, W, Y",
+        5: "K",
+        8: "J, X",
+        10: "Q, Z"
+    }
+    
+    letter_points = {}
+    for points, letters in point_values.items():
+        for letter in letters:
+            letter_points[letter] = points
+    
+    # Calculate score by sum of points
+    score = 0
+    for letter in word:
+        score += letter_points[letter]
+    
+    # Add bonus points for words 7-10 letters long
+    if 7 <= len(word) <= 10:
+        score += 8
+    
+    return score
 
 def get_highest_word_score(word_list):
-    pass
\ No newline at end of file
+
+    if not word_list:
+        return None
+    
+    word_positions = {}
+    for i in range(len(word_list)):
+        word = word_list[i]
+        if word not in word_positions:
+            word_positions[word] = i
+    
+    # Dctionary that stores each words metadata
+    word_metadata = {}
+    for word in word_list:
+        word_metadata[word] = {
+            'score': score_word(word),
+            'length': len(word),
+            'position': word_positions[word]
+        }
+    
+    # Find the highest score
+    highest_score = 0
+    for word, metadata in word_metadata.items():
+        if metadata['score'] > highest_score:
+            highest_score = metadata['score']
```

Comment:
When we have to start looping through the items in a dictionary, it can be a good idea to try and find a different approach!